### PR TITLE
[5.0] Restrict Request::only() to create non-exist keys

### PR DESCRIPTION
**Pull request by MalikinSergey**
Restrict Request::only() to create array keys that not exists in input

Only doubt is it bugfix or change?

### CODE REVIEW

File: src/Illuminate/Http/Request.php [hunk 1]
```diff
@@ -307,7 +307,9 @@ public function only($keys)
 
 		foreach ($keys as $key)
 		{
-			array_set($results, $key, array_get($input, $key));
+			if(($value = array_get($input, $key)) !== null){
```
Comment (GrahamCampbell) [inline src/Illuminate/Http/Request.php]: cs

**Comment by GrahamCampbell** [comment]
Tests would be really useful here.

File: src/Illuminate/Http/Request.php [hunk 2]
```diff
@@ -307,7 +307,11 @@ public function only($keys)
 
 		foreach ($keys as $key)
 		{
-			array_set($results, $key, array_get($input, $key));
+			$value = array_get($input, $key);
+			
+			if( ! is_null($value)){
```
Comment (GrahamCampbell) [inline src/Illuminate/Http/Request.php]: Still not quite right. Take a look at our other if statements. :)

**Comment by MalikinSergey** [comment]
As I see in HttpRequestTest::testOnlyMethod, initial behavior which creating new null array keys - is expected behavior.

```php
		$this->assertEquals(array('developer' => array('name' => 'Taylor'), 'test' => null), $request->only('developer.name', 'test'));
```

**Comment by taylorotwell** [comment]
Passing null is fine. Null is sometimes sent with JSON APIs. I don't see why it should be removed.

**Comment by MalikinSergey** [comment]
@taylorotwell By the way, logic of `Request::only()` is not matches with `array_only()` (as might think from names) - `array_only()` returns only existing key-values
So, sometimes have to use ` $model->update( array_only(\Request::all(), ['title','content','status'] ) ) ` for example, to prevent  writing `null` values of title, content  to `$model`, when sent only `status` field from sepatated view `<form>` to `ResourceController@postUpdate` controller

**Pull request closed without merging**